sgxvm/src/precompiles/compliance_bridge.rs

```diff
@@ -45,119 +47,411 @@ impl LinearCostPrecompileWithQuerier for ComplianceBridge {
+                Some(addr) => addr,
+                None => {
+                    return Err(PrecompileFailure::Revert {
+                        exit_status: ExitRevert::Reverted,
+                        output: encode(&vec![AbiToken::String(
+                            "invalid user address".into(),
+                        )]),
+                    });
+                }
+            };
+
+            let encoded_request = coder::encode_get_verification_data(user_address, caller);
+
+            match querier::make_request(querier, encoded_request) {
+                Some(result) => {
+                    let get_verification_data = protobuf::parse_from_bytes::<ffi::QueryGetVerificationDataResponse>(result.as_slice())
                         .map_err(|_| PrecompileFailure::Revert {
                             exit_status: ExitRevert::Reverted,
-                            output: encode(&vec![AbiToken::String("cannot parse protobuf response".into())])
+                            output: encode(&vec![AbiToken::String(
+                                "cannot decode protobuf response".into(),
+                            )]),
                         })?;
 
-                    Ok((ExitSucceed::Returned, Vec::default()))
-                },
+                    let data = get_verification_data
+                        .data
+                        .into_iter()
+                        .flat_map(|log| {
+                            let issuer_address = Address::from_slice(&log.issuerAddress);
+                            let tokens = vec![AbiToken::Tuple(vec![
+                                AbiToken::Address(issuer_address.clone()),
+                                AbiToken::String(log.originChain.clone()),
+                                AbiToken::Uint(log.issuanceTimestamp.into()),
+                                AbiToken::Uint(log.expirationTimestamp.into()),
+                                AbiToken::Bytes(log.originalData.clone().into()),
+                                AbiToken::String(log.schema.clone()),
+                                AbiToken::String(log.issuerVerificationId.clone()),
+                                AbiToken::Uint(log.version.into()),
+                            ])];
+
+                            tokens.into_iter()
+                        })
+                        .collect::<Vec<AbiToken>>();
+
+                    let encoded_response = encode(&[AbiToken::Array(data)]);
+                    Ok((ExitSucceed::Returned, encoded_response.to_vec()))
+                }
                 None => {
-                    return Err(PrecompileFailure::Revert {
+                    Err(PrecompileFailure::Revert {
                         exit_status: ExitRevert::Reverted,
-                        output: encode(&vec![AbiToken::String("call to x/compliance failed".into())])
+                        output: encode(&vec![AbiToken::String(
+                            "call to x/compliance failed".into(),
```

Comment:
specify exact error, this string is the same in multiple places, will have a hard time debugging

Reply:
Added the function name to the error string.